conan/cli/commands/build.py

```diff
@@ -29,6 +29,8 @@ def build(conan_api, parser, *args):
                         help="Deployer output folder, base build folder by default if not set")
     parser.add_argument("--build-require", action='store_true', default=False,
                         help='Whether the provided path is a build-require')
+    parser.add_argument("--envs-generation", default=None, choices=["never"],
```

Comment:
Lets change it to something "softer", like:
- ``--envs-generation=false`` => No VirtualXXXEnv generated, current proposal
- ``--envs-generation=never`` => No script at all, not even ``conanvcvars.bat`` (for the future)